### PR TITLE
Coverage improvement matchers and equality

### PR DESCRIPTION
**Pull request by juanlavagnini**

### CODE REVIEW

**Comment by TRICOT-Hugo** [comment]
This should be merge in issue/5-coverage-improvement rather than main. Please change before a review is made.

File: src/main/java/org/mockito/internal/matchers/Matches.java [hunk 1]
```diff
@@ -23,6 +26,8 @@ public Matches(Pattern pattern) {
 
     @Override
     public boolean matches(Object actual) {
+        Coverage.setTotalBranches("Matches::matches", 1);
+        Coverage.reached("Matches::matches", 0);
         return (actual instanceof String) && pattern.matcher((String) actual).find();
```
Comment (TRICOT-Hugo) [inline src/main/java/org/mockito/internal/matchers/Matches.java:31]: It's a tricky one, I don't think you can do better, but technically the `&&` implies another brnch.

**Review by TRICOT-Hugo — COMMENTED** [review]
Changing the branch still kept the other commits, I don't get why, there should only be yours.

File: src/main/java/org/mockito/internal/matchers/Matches.java (hunk 1 above)
Comment (juanlavagnini) [reply src/main/java/org/mockito/internal/matchers/Matches.java:31]: It's posible to change the code in order to make a proper coverage analysis, but i think this it the best option (i'm not so happy with that)

**Pull request merged**